Check links one time per week

RaheemAmer opened the pull request:
## Reason
This PR is made on #9's suggestion

## Source
- https://dev.to/thibaultduponchelle/check-markdown-links-with-github-action-3c2m

## Feedback
Happy to hear and discuss any feedback, cheers ❤️

RaheemAmer committed:
Create checklinks.yml

RaheemAmer committed:
Create mlc_config.json

eddiejaoude commented on .github/workflows/checklinks.yml:
```diff
@@ -0,0 +1,13 @@
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
```
`main` is the default branch, however release `v3` is recommended to use https://github.com/actions/checkout

RaheemAmer replied:
Thank you so much, I have a question on the v3, when you search for the latest version, it shows  v2.4.2, but when I choose any v3 it tells me to use the latest version instead, so what do you recommend me to do?

## Preview (1)
![image](https://user-images.githubusercontent.com/18606136/191462197-9ba3bdef-b0fb-4bf2-8f93-ecb3aa250e8e.png)

## Preview (2)
![image](https://user-images.githubusercontent.com/18606136/191461754-5d682491-8ca1-4e96-90f3-3c63a1201f72.png)

eddiejaoude replied:
I think any release is better than using a branch

eddiejaoude commented on .github/workflows/checklinks.yml:
```diff
@@ -0,0 +1,13 @@
+
+on:
+  schedule:
+  - cron: "0 13 * * 1"
```
Probably best to run this on use trigger rather than the cron, so the user can get immediate feedback

Something like
```yaml
on:
   push 
```

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push

EmmaDawsonDev replied:
I think the idea is to check through all the links once a week hence the cron, but maybe a trigger on push would be useful as well.

RaheemAmer replied:
@eddiejaoude @EmmaDawsonDev 
Definitely a very good suggestion  👍🏻

eddiejaoude commented on .github/workflows/checklinks.yml:
```diff
@@ -0,0 +1,13 @@
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      continue-on-error: true
```
I couldn't see this option in the docs? https://github.com/gaurav-nelson/github-action-markdown-link-check#custom-variables

RaheemAmer replied:
@eddiejaoude most of the PR is based on this repo, that's why this line exists
- https://github.com/gaurav-nelson/github-action-markdown-link-check

## Preview
![image](https://user-images.githubusercontent.com/18606136/191464616-55bf8aa9-2adf-47cf-b5b2-5e8287820f83.png)

RaheemAmer replied:
@eddiejaoude 
You're right, it's from this article
- https://dev.to/thibaultduponchelle/check-markdown-links-with-github-action-3c2m

## Preview on the reason why this line exists
![image](https://user-images.githubusercontent.com/18606136/191465637-08e545a2-e970-4234-88a2-8f4c53e46e52.png)

eddiejaoude replied:
Not sure if it is still available on the Action, might need to dig deeper.

If it is available, it is probably best for @EmmaDawsonDev to decide if she prefers failure or report only

eddiejaoude commented on mlc_config.json:
```diff
@@ -0,0 +1,6 @@
+// 429 Too Many Requests
+// This is the typical github answer after too much requests, so I would like to ignore this status when it happens.
```
Great to have comments 👍  but I think this will cause the file to be invalid

RaheemAmer replied:
@eddiejaoude 
My bad, you're definitely right 👍🏻

eddiejaoude reviewed (changes requested):
Interesting GitHub Action, I didn't know about it, so thank you 👍 .

I added a few comments, any questions let me know 🤓

EmmaDawsonDev reviewed (commented):
Thanks for your contribution @RaheemAmer . I'm not too familiar with github actions yet so I'm curious to know where the results of this action end up so that I can action any links that no longer work.

RaheemAmer commented:
> Thanks for your contribution @RaheemAmer . I'm not too familiar with github actions yet so I'm curious to know where the results of this action end up so that I can action any links that no longer work.

My pleasure ❤️  @EmmaDawsonDev , It's gonna produce something similar to this
- https://github.com/RaheemAmer/Self-study/actions/runs/3095178862/jobs/5009319888

## Preview
![image](https://user-images.githubusercontent.com/18606136/191458596-d854c967-931a-4835-9730-1c3a5b64c812.png)

RaheemAmer commented:
> Interesting GitHub Action, I didn't know about it, so thank you 👍 .
> 
> I added a few comments, any questions let me know 🤓

You're most welcome ❤️ @eddiejaoude  and also thank you so much for your comments and review, I'll recheck your comments and start working on the PR

RaheemAmer committed:
Fixed few issues 

## Changes

- From using cron to using push
- Changed branch from master to main
- Updated the version of the checkout action into the latest version available 
- Added quiet and verbose mode  to offer more detailed info, [Check Custom variables Table](https://github.com/gaurav-nelson/github-action-markdown-link-check#custom-variables)
- Added config file path, forgot to add it

RaheemAmer committed:
Removed Comments

## Reason

- May cause the file to be invalid

RaheemAmer commented:
I've done all the changes except changing into using checkout rather than a branch, I'm having a little bit of confusion on what to write exactly, so any help will be more than welcome, other than this everything is crystal clear
@eddiejaoude 

and @EmmaDawsonDev Could you please inform me whether you want or prefers failure or report only regarding this issue https://github.com/AccessibleForAll/Resources/pull/16#discussion_r976266213

Thank you so much for your time, effort and cooperation, and I'm sorry if I did anything wrong. I'm still learning ❤️

CBID2 reviewed (commented):
Looks good to me @RaheemAmer!
